src/OpenTelemetry.Extensions.EventSource/OpenTelemetryEventSourceLogEmitter.cs

```diff
@@ -0,0 +1,224 @@
+            this.shouldListenToFunc = shouldListenToFunc;
+
+            var logEmitter = this.openTelemetryLoggerProvider.CreateEmitter();
+            Debug.Assert(logEmitter != null, "logEmitter was null");
+
+            this.logEmitter = logEmitter!;
+
+            lock (this.lockObj)
+            {
+                foreach (EventSource eventSource in this.eventSourcesBeforeConstructor)
+                {
+                    this.ProcessSource(eventSource);
+                }
+
+                this.eventSourcesBeforeConstructor = null;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Dispose()
+        {
+            foreach (EventSource eventSource in this.eventSources)
+            {
+                this.DisableEvents(eventSource);
+            }
+
+            this.eventSources.Clear();
+
+            if (this.disposeProvider)
+            {
+                this.openTelemetryLoggerProvider.Dispose();
+            }
+
+            base.Dispose();
+        }
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+        /// <inheritdoc/>
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            Debug.Assert(eventSource != null, "EventSource was null.");
+
+            try
+            {
+                if (this.eventSourcesBeforeConstructor != null)
+                {
+                    lock (this.lockObj)
+                    {
+                        if (this.eventSourcesBeforeConstructor != null)
+                        {
+                            this.eventSourcesBeforeConstructor.Add(eventSource!);
+                            return;
+                        }
+                    }
+                }
+
+                this.ProcessSource(eventSource!);
+            }
+            finally
+            {
+                base.OnEventSourceCreated(eventSource);
+            }
+        }
+#pragma warning restore CA1062 // Validate arguments of public methods
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+        /// <inheritdoc/>
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            Debug.Assert(eventData != null, "EventData was null.");
+
+            string? rawMessage = eventData!.Message;
+
+            LogRecordData data = new(Activity.Current)
+            {
+#if NETSTANDARD2_1_OR_GREATER
+                Timestamp = eventData.TimeStamp,
+#endif
+                EventId = new EventId(eventData.EventId, eventData.EventName),
+                LogLevel = ConvertEventLevelToLogLevel(eventData.Level),
+            };
+
+            LogRecordAttributeList attributes = default;
+
+            attributes.Add("event_source.name", eventData.EventSource.Name);
+
+            if (eventData.ActivityId != Guid.Empty)
+            {
+                attributes.Add("event_source.activity_id", eventData.ActivityId);
```

Comment:
😄 The "Activity" from EventSource and "Activity" for Span can be confusing. we can add some wording in the readme.md later.